pallets/dapp-staking/src/migration.rs

```diff
@@ -412,3 +417,111 @@ pub mod v6 {
         pub period: PeriodNumber,
     }
 }
+
+const PALLET_MIGRATIONS_ID: &[u8; 16] = b"dapp-staking-mbm";
+
+pub struct LazyMigration<T, W: WeightInfo>(PhantomData<(T, W)>);
+
+impl<T: Config, W: WeightInfo> SteppedMigration for LazyMigration<T, W> {
+    type Cursor = <T as frame_system::Config>::AccountId;
+    // Without the explicit length here the construction of the ID would not be infallible.
+    type Identifier = MigrationId<16>;
+
+    /// The identifier of this migration. Which should be globally unique.
+    fn id() -> Self::Identifier {
+        MigrationId {
+            pallet_id: *PALLET_MIGRATIONS_ID,
+            version_from: 0,
+            version_to: 1,
+        }
+    }
+
+    fn step(
+        mut cursor: Option<Self::Cursor>,
+        meter: &mut WeightMeter,
+    ) -> Result<Option<Self::Cursor>, SteppedMigrationError> {
+        let required = W::step();
+        // If there is not enough weight for a single step, return an error. This case can be
+        // problematic if it is the first migration that ran in this block. But there is nothing
+        // that we can do about it here.
+        if meter.remaining().any_lt(required) {
+            return Err(SteppedMigrationError::InsufficientWeight { required });
```

Comment:
Quick question: What happens if this error is emitted? I assume nothing, and no storage rollback is required since nothing has changed, but will the migration be retried on the next block?

Reply:
tests will catch this